arrayfire_wrapper/_backend.py

```diff
@@ -64,40 +70,85 @@ def _get_backend_path_config() -> _BackendPathConfig:
+
 
         if cuda_path and (cuda_path / "bin").is_dir() and (cuda_path / "nvvm/bin").is_dir():
             cuda_found = True
 
-        return _BackendPathConfig("", ".dll", af_path, cuda_found)
+        return _BackendPathConfig("", ".dll", af_path, af_is_user_path, cuda_found)
 
     if platform_name == _SupportedPlatforms.darwin.value:
         default_cuda_path = Path("/usr/local/cuda/")
 
         if not af_path:
             af_path = _find_default_path("/opt/arrayfire", "/usr/local")
+            try:
+                af_path = _find_default_path(f"C:/Program Files/ArrayFire/v{ARRAYFIRE_VER_MAJOR}",
+                                              "C:/Program Files (x86)/ArrayFire/v{ARRAYFIRE_VER_MAJOR}")
+            except ValueError:
+                af_path = None
 
         if not (cuda_path and default_cuda_path.exists()):
             cuda_found = (default_cuda_path / "lib").is_dir() and (default_cuda_path / "/nvvm/lib").is_dir()
 
-        return _BackendPathConfig("lib", f".{ARRAYFIRE_VER_MAJOR}.dylib", af_path, cuda_found)
+        return _BackendPathConfig("lib", f".{ARRAYFIRE_VER_MAJOR}.dylib", af_path, af_is_user_path, cuda_found)
 
     if platform_name == _SupportedPlatforms.linux.value:
         default_cuda_path = Path("/usr/local/cuda/")
 
         if not af_path:
-            af_path = _find_default_path(f"/opt/arrayfire-{ARRAYFIRE_VER_MAJOR}", "/opt/arrayfire/", "/usr/local/")
+            try:
+                af_path = _find_default_path(f"/opt/arrayfire-{ARRAYFIRE_VER_MAJOR}", "/opt/arrayfire/", "/usr/local/")
+            except ValueError:
+                af_path = None
 
         if not (cuda_path and default_cuda_path.exists()):
             if "64" in platform.architecture()[0]:  # Check either is 64 bit arch is selected
                 cuda_found = (default_cuda_path / "lib64").is_dir() and (default_cuda_path / "nvvm/lib64").is_dir()
             else:
                 cuda_found = (default_cuda_path / "lib").is_dir() and (default_cuda_path / "nvvm/lib").is_dir()
 
-        return _BackendPathConfig("lib", f".so.{ARRAYFIRE_VER_MAJOR}", af_path, cuda_found)
+        return _BackendPathConfig("lib", f".so.{ARRAYFIRE_VER_MAJOR}", af_path, af_is_user_path, cuda_found)
 
     raise OSError(f"{platform_name} is not supported.")
 
+# finds paths to locally packaged arrayfire libraries if they exist in site
+def _find_site_local_path() -> Path:
+    local_paths = ["."]
+
+    # module search paths
+    af_module = __import__(__name__)
+    module_paths = af_module.__path__ if af_module.__path__ else []
+    for path in module_paths:
+        local_paths.append(path)
+
+    # site search path
+    purelib_path = sysconfig.get_path('purelib')
+    platlib_path = sysconfig.get_path('platlib')
+    local_paths.append(purelib_path)
+    local_paths.append(platlib_path)
+
+    # sys search path
+    local_paths.extend(sys.path)
+
+    module_name = af_module.__name__
+    for path in local_paths:
```

Comment:
```
# Its a bad practice to copy multiple times a str variable that can be changed any time. Lets define it once and reuse later
binaries_path = Path(af_module.__name__) / "binaries"

for path in local_paths:
    # Assuming you applied changes in the comments above, we can avoid extra tabulation by exiting the cycle earlier
    if not lpath.exists():
        continue

    files = [x.name for x in binaries_path.glob("*") if x.is_file()]  # can be simplified 'coz binaries_path is already a path
    # It'd be better to bind lib names with their devices/frameworks because likely adding a new backend, e.g., cuda2 will lead to adding afcuda2 or smth
    query_libnames = [f'af{backend.name}' for backend in BackendType] + ['af', 'forge']  # yup, it will also add afunified, but who cares if it will be skipped 🤷 
    found_lib_in_dir = any(lib_name in files for lib_name in query_libnames)  # lets make it O(n) at least

    if not found_lib_in_dir:  # Typical trick to avoid extra tabulation
        continue

    if VERBOSE_LOADS:
        print(f"Found binary file by path: {path / binaries_path}")
    
    return path / binaries_path  # returns Path
 
raise RuntimeError("No binaries were found in site path.")
 ```

Reply:
Yea, and it's better to make descriptive comments if you catch errors or add implicit verbosing